assignment3/assignment3.py

```diff
@@ -0,0 +1,30 @@
+class Dictionary():
+    def __init__(self, wordsList):
+        self.words = list()
+        
+    def isWord(self, string):
+        return string in self.words
+    def isPrefix(self,string,words)
+        for x in self.words
+            if(string == words[x])
+                return True
+        return False
+    
+    def wordSearch(row,cols,matrixChar,dictionary):
+        # For searching in all 8 direction
+        dx = [-1, -1, -1, 0, 0, 1, 1, 1 ];
+        dy = [-1, 0, 1, -1, 1, -1, 0, 1 ];
+        found = []
+        for i in range(0,rows):
+            for j in range(0,cols):
+                    pref = matrix[i][j]
+                    if(isPrefix(pref)):
+                        for k in dx:
+                            idx = i+dx[i];
+                            jdy = j+dy[j];
+                            if(isWord(idx, jdy):
+                                wordSearch(matrixChar[idx],matrixChar[jdy],matrixChar,dictionary)
```

Comment:
For the structure of recursion to be correct, you should check the return value here and return `True` if the recursive call successfully located the word somewhere down the branch.

There also needs to be a check somewhere (before the for loops start) whether the currently investigated branch has already located the word or determined that the word can not be found to terminate the search. That result is what then needs to trickle back up the recursion.

It may be helpful to write the first layer of recursion as a separate function, at least initially. Then `startWordSearch` can set up all the recursion parameters properly and call the recursive `wordSearch` which would be able to rely on the neatly set up parameters from the entry function.